bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/FileUtil.java

```diff
@@ -302,4 +313,94 @@ public static void pipe(InputStream input, OutputStream output) throws IOExcepti
 		}
 	}
 
+	/**
+	 * Reads the file contents from the project.
+	 * @param project The project to read file from
+	 * @param relativeFilePath Relative path to project leading to a file
+	 * @return String contents for the read file
+	 */
+	public static String readFile(IProject project, String relativeFilePath)
+	{
+		StringBuilder fileContents = new StringBuilder();
+		IFile filePath = project.getFile(new Path(relativeFilePath));
+		if (!filePath.exists())
+		{
+			Logger.log(MessageFormat.format("{0} couldn't find", filePath.toString())); //$NON-NLS-1$
+			return null;
+		}
+		
+		try
+		{
+			InputStream inputStream = filePath.getContents(true);
+			Scanner scanner = new Scanner(inputStream);
+			while (scanner.hasNext())
+			{
+				fileContents.append(scanner.nextLine());
+				fileContents.append(System.getProperty("line.separator"));
+			}
+			inputStream.close();
+			scanner.close();
+		}
+		catch (Exception e)
+		{
+			Logger.log(e);
+			return null;
+		}
+
+		return fileContents.toString();
```

Comment:
We can replace this code with call readFile(String absoluteFilePath) as you have already implemented this

Reply:
Done